pkg/minikube/minikube.go

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2021 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package minikube
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"knative.dev/kn-plugin-quickstart/pkg/install"
+)
+
+var clusterName = "minikube-knative"
+var minikubeVersion = "v1.23"
+
+// SetUp creates a local Minikube cluster and installs all the relevant Knative components
+func SetUp() error {
+	if err := createMinikubeCluster(); err != nil {
+		return fmt.Errorf("creating cluster: %w", err)
+	}
+	if err := install.Serving(); err != nil {
+		return fmt.Errorf("install serving: %w", err)
+	}
+	if err := install.Kourier(); err != nil {
+		return fmt.Errorf("install kourier: %w", err)
+	}
+	if err := install.KourierMinikube(); err != nil {
+		return fmt.Errorf("configure kourier: %w", err)
+	}
+	if err := install.Eventing(); err != nil {
+		return fmt.Errorf("install eventing: %w", err)
+	}
+	return nil
+}
+
+func createMinikubeCluster() error {
+	if err := checkMinikubeVersion(); err != nil {
+		return fmt.Errorf("minikube version: %w", err)
+	}
+	if err := checkForExistingCluster(); err != nil {
+		return fmt.Errorf("existing cluster: %w", err)
+	}
+	return nil
+}
+
+// checkMinikubeVersion validates that the user has the correct version of Minikube installed.
```

Comment:
what's the correction version of minikube?  I don't think we want to hard code to only version `1.23` I think we ask for a minimum version like 1.23+

Reply:
Sounds good ... we'll also need to update the kind option to do the same (I'll open an issue for that).